datastore.py

```diff
@@ -0,0 +1,35 @@
+import boto3
+import json
+
+from botocore.errorfactory import ClientError
+
+s3 = boto3.resource('s3')
+
+class Datastore():
+    """Datastore allows us to persist some data outside the application.
+    
+    The data property is loaded initially from S3 and then cached for the
+    lifetime of the application (subsequent reads will not pull from S3).

```

Comment:
If the application only reads from the S3 bucket once, what happens if 2 people make changes to the same item in a short period of time?  Would you have to restart in order to clear the cache and get the latest version of the data?  Or does it not matter in this case?

Reply:
We only have one instance of bot running so it shouldn't matter. Any "writes" only occur on the local in-memory data so it would be correct. 

IF we had multiple bots running though, you could have bot1 update its local data and then bot2 comes online and reads the stale data.

Reply:
it also means, if the bot crashes without calling `save()` then those changes could be lost.

Reply:
Makes sense, thanks!